airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -57,6 +60,21 @@ class KubernetesHook(BaseHook):
     conn_name_attr = 'kubernetes_conn_id'
     default_conn_name = 'kubernetes_default'
     conn_type = 'kubernetes'
+    hook_name = 'Kubernetes Cluster Connection'
+
+    @staticmethod
+    def monkey_patch_connection_form(form_to_patch):
```

Comment:
Why do we need a monkeypatch? There is no other way to do this without it?

Reply:
We don't. I rewrote it (looking at your example) and we do not use monkeypatching now from outside, we still add attributes to ConnectionForm class, though but this is far more controlled now (each providers provide a dictionary of fields). So no monkey-patching from the provider code - instead we only add fields based on their names in the Connnection Form class itself.,